File: scripts/compare_sitemap.js
const fs = require('fs');
const path = require('path');
const root = process.cwd();
const pagesDir = path.join(root, 'pages');
const exclude = new Set(['_app.jsx', '_document.jsx', '404.jsx', 'sitemap.xml.js', 'robots.txt.js']);

function collectPageRoutes(dir) {
  const routes = [];
  for (const name of fs.readdirSync(dir)) {
    const full = path.join(dir, name);
    const stat = fs.statSync(full);
    if (stat.isDirectory()) {
      routes.push(...collectPageRoutes(full));
      continue;
    }
    if (!name.endsWith('.jsx') && !name.endsWith('.js')) continue;
    const rel = path.relative(pagesDir, full).replace(/\\/g, '/');
    if (exclude.has(rel)) continue;
    if (rel.startsWith('api/') || rel.startsWith('admin/')) continue;
    if (/\[.*\]/.test(rel)) continue;
    let route = '/' + rel.replace(/\.jsx$|\.js$/, '');
    route = route.replace(/\/index$/, '');
    if (route === '') route = '/';
    routes.push(route);
  }
  return routes;
}

const pageRoutes = [...new Set(collectPageRoutes(pagesDir))].sort();
const sitemapText = fs.readFileSync(path.join(pagesDir, 'sitemap.xml.js'), 'utf8');
const sitemapUrls = [...new Set(
  [...sitemapText.matchAll(/loc\s*:\s*['\"]https:\/\/weoneaviation\.in([^'\"]*)['\"]/g)].map(m => m[1] === '' ? '/' : m[1])
)].sort();

const missing = pageRoutes.filter(r => !sitemapUrls.includes(r));
const extra = sitemapUrls.filter(u => !pageRoutes.includes(u));
console.log('PAGE_COUNT', pageRoutes.length);
console.log('SITEMAP_COUNT', sitemapUrls.length);
console.log('MISSING_IN_SITEMAP');
missing.forEach(r => console.log(r));
console.log('EXTRA_IN_SITEMAP');
extra.forEach(u => console.log(u));

File: scripts/compare_sitemap.py
import os
import re

root = os.getcwd()
pages_dir = os.path.join(root, 'pages')
exclude = {'_app.jsx', '_document.jsx', '404.jsx', 'sitemap.xml.js', 'robots.txt.js'}

page_routes = []
for dirpath, dirnames, filenames in os.walk(pages_dir):
    for filename in filenames:
        if not filename.endswith(('.jsx', '.js')):
            continue
        rel = os.path.relpath(os.path.join(dirpath, filename), pages_dir).replace('\\', '/')
        if rel in exclude:
            continue
        if rel.startswith('api/') or rel.startswith('admin/'):
            continue
        if re.search(r'\[.*\]', rel):
            continue
        route = '/' + re.sub(r'(\.jsx|\.js)$', '', rel)
        route = re.sub(r'/index$', '', route)
        if route == '':
            route = '/'
        page_routes.append(route)
page_routes = sorted(set(page_routes))

sitemap_path = os.path.join(pages_dir, 'sitemap.xml.js')
text = open(sitemap_path, 'r', encoding='utf-8').read()
urls = re.findall(r"loc\s*:\s*['\"]https://weoneaviation\.in([^'\"]*)['\"]", text)
urls = ['/' if u == '' else u for u in urls]
urls = sorted(set(urls))

print('PAGE_COUNT', len(page_routes))
print('SITEMAP_COUNT', len(urls))
print('MISSING_IN_SITEMAP')
for p in page_routes:
    if p not in urls:
        print(p)
print('EXTRA_IN_SITEMAP')
for u in urls:
    if u not in page_routes:
        print(u)
